lib/rules/wrap-iife.js

```diff
@@ -46,64 +56,96 @@ module.exports = {
                 nextToken && nextToken.value === ")";
         }
 
-        return {
+        /**
+        * Get the function node from an IIFE
+        * @param {ASTNode} node to evaluate
+        * @returns {ASTNode} node that is the function expresion of the given IIFE, or null if non exist
+        */
+        function getFunctionNodeFromIIFE(node) {
+            if (node.type !== "CallExpression") {
+                return null;
+            }
+
+            const callee = node.callee;
+
+            if (callee.type === "FunctionExpression") {
+                return callee;
+            }
+
+            if (includeFunctionPrototypeMethods &&
+                callee.type === "MemberExpression" &&
+                callee.object.type === "FunctionExpression" &&
+                callee.property.type === "Identifier" &&
+                (callee.property.name === "call" || callee.property.name === "apply")
```

Comment:
I think this needs to consider computed properties (e.g. `function(){}[call]()` and `function(){}["call"]()`).
Maybe:

```js
            if (includeFunctionPrototypeMethods &&
                callee.type === "MemberExpression" &&
                callee.object.type === "FunctionExpression" &&
                (astUtils.getStaticPropertyName(callee) === "call" || astUtils.getStaticPropertyName(callee) === "apply")